pkg/objectstate/rte/nodegroup.go

```diff
@@ -80,11 +89,27 @@ func (obj nodeGroupFinder) FindState(mf Manifests, tree nodegroupv1.Tree) []obje
 		HyperShiftNodePoolLabel: poolName,
 	}
 
+	var desiredConfigMap *corev1.ConfigMap
+	existingConfigMap, ok := obj.em.configMaps[generatedName]
+	errFormat := fmt.Errorf("failed to find config map %s/%s", mf.Core.DaemonSet.Namespace, generatedName)
+	// a case where kubelet controller is not creating the ConfigMap, is undesired but yet possible.
+	// this is why the operator should log here
+	// and not bubble up the error that would interfere with its normal operation.
+	switch {
+	case !ok:
+		klog.ErrorS(errFormat, "")
+	case existingConfigMap.configMapError != nil:
+		klog.ErrorS(existingConfigMap.configMapError, errFormat.Error())
+	default:
+		desiredConfigMap = existingConfigMap.configMap
```

Comment:
do we need a DeepCopy here?

Reply:
No. We're not mutating the config map.
All the controller is doing is exercise a hash function on the data